grunt_tasks/plugin.js

```diff
@@ -150,27 +193,126 @@ module.exports = function (grunt) {
                     }
                 }
             });
+
+            // If the plugin config has no webpack section, no defaultLoaders
+            // property in the webpack section, or the defaultLoaders is
+            // explicitly set to anything besides false, then augment the
+            // webpack loader configurations with the plugin source directory.
+            if (!config.webpack || config.webpack.defaultLoaders === undefined || config.webpack.defaultLoaders !== false) {
+                var numLoaders = grunt.config.get('webpack.options.module.loaders').length;
+                for (var i = 0; i < numLoaders; i++) {
+                    var selector = 'webpack.options.module.loaders.' + i + '.include';
+                    var loaders = grunt.config.get(selector);
+                    grunt.config.set(selector, loaders.concat([path.resolve(dir)]));
+                }
+            }
+
+            var newConfig = webpackHelper(grunt.config.get('webpack.options'), helperConfig);
+            grunt.config.set('webpack.options', newConfig);
         }
 
-        function addDependencies(deps) {
+        grunt.registerTask('npm-install', 'Install plugin NPM dependencies', function (plugin, localNodeModules) {
+            // To attach the stream listeners, we need this to be an
+            // asynchronous child process launch.
+            var done = this.async();
+
+            // Start building the list of arguments to the NPM executable.
+            //
+            // We want color output embedded in the Grunt output.
+            var args = ['--color=always'];
+
+            // If the plugin requested to install the dependencies in its own
+            // dedicated directory, set the prefix option.
+            if (localNodeModules === 'true') {
+                args = args.concat(['--prefix', path.resolve('node_modules_' + plugin)]);
+            }
+
+            // Get the list of the packages to install and append them to the
+            // args object.
+            var modules = Array.prototype.slice.call(arguments, 2);
+            args = args.concat(['install'], modules);
+
+            // Formulate path to npm executable.
+            var npm = path.resolve(path.dirname(process.argv[0]), 'npm');
+
+            // Launch the child process.
+            var child = child_process.spawn(npm, args);
```

Comment:
I think we can do this synchronously and not have to mess with the child pipes if we do:

    child_process.execSync(command, {stdio: 'inherit'});

Documented [here](https://nodejs.org/api/child_process.html#child_process_options_stdio).